### PR TITLE
fix(anchor): ink correct switch direction

### LINKED ISSUE #41246: Anchor component CSS breaks when switching the direction between horizontal and vertical
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/dynamic-anchor-direction-ngtk7d?file=/demo.tsx)

### Steps to reproduce

Click and anchor tag. Adjust the viewport (wider) to render the anchor tags vertically. Adjust the viewport (narrower) to render the anchor tags horizontally.

### What is expected?

The CSS will be correct when switching the direction between horizontal and vertical.

### What is actually happening?

The CSS is off.

| Environment | Info |
| --- | --- |
| antd | 5.3.1 |
| React | 18 |
| System | Windows 11 |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by acyza**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #41246 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  ink correct switch direction |
| 🇨🇳 Chinese | ink正确切换方向 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/anchor/Anchor.tsx [hunk 1]
```diff
@@ -176,9 +176,13 @@ const AnchorContent: React.FC<InternalAnchorProps> = (props) => {
       if (anchorDirection !== 'horizontal') {
         spanLinkNode.current.style.top = `${linkNode.offsetTop + linkNode.clientHeight / 2}px`;
         spanLinkNode.current.style.height = `${linkNode.clientHeight}px`;
+        spanLinkNode.current.style.left = '';
+        spanLinkNode.current.style.width = '';
       } else {
         spanLinkNode.current.style.left = `${linkNode.offsetLeft}px`;
         spanLinkNode.current.style.width = `${linkNode.clientWidth}px`;
+        spanLinkNode.current.style.top = '';
+        spanLinkNode.current.style.height = '';
```
Comment (yoyo837) [inline components/anchor/Anchor.tsx:185]: Can the two if statements be combined to reduce the lines of code?

Comment (acyza) [reply components/anchor/Anchor.tsx:185]: 好像不太好合并，或许可以增加更多的三目运算，但单行过长也需要换行，最终行数也差不了多少。

Comment (acyza) [reply components/anchor/Anchor.tsx:185]: 图片

Comment (yoyo837) [reply components/anchor/Anchor.tsx:185]: 那就为 spanLinkNode.current.style 提取临时变量吧

**Review by afc163 — APPROVED** [review]

**Pull request merged**